Bug 1999314: Resync all controllers periodically

jhadvig opened the pull request:
Resync of the controllers was removed in https://github.com/openshift/console-operator/pull/531. But since it looks like we need it Im adding them back.
Together with https://github.com/openshift/console-operator/pull/580 PR this issue should be solved. Wondering if a minute is the right period here.

/assign @spadgett 
@TheRealJon @florkbr FIY

jhadvig commented:
/bugzilla refresh

spadgett commented:
@jhadvig Was the issue that originally caused us to rollback the resync change fixed?

https://bugzilla.redhat.com/show_bug.cgi?id=1948524

jhadvig commented:
@spadgett so the issue was with the downloads deployment which was being rolled out every minute. That was cause by missing, which was due to the fact that we have been missing calling method for [updating deployment generation](https://github.com/jhadvig/console-operator/blob/34940460b4721db3fa780d14e3ae26579f5dfa2a/pkg/console/controllers/downloadsdeployment/controller.go#L119) which we added. Unfortunately with that PR I've removed the resync methods call.

Tested the change and there should be no redeployments happening.

spadgett reviewed (approved):
/lgtm

jhadvig commented:
/cherry-pick release-4.9
